torch/testing/_internal/distributed/distributed_test.py

```diff
@@ -4349,6 +4350,57 @@ def _test_DistributedDataParallel_SyncBatchNorm(
             )
             self._barrier()
 
+        @skip_if_lt_x_gpu(2)
+        @unittest.skipIf(
+            BACKEND != "nccl" and BACKEND != "gloo",
+            "Only NCCL and GLOO backend support DistributedDataParallel",
+        )
+        def test_post_localSGD_optimizer_parity(self, grad_is_view=False):
+            learning_rate = 0.03
+            period = 4
+            warmup_steps = 10
+            torch.cuda.set_device(self.rank)
+            net = torch.nn.parallel.DistributedDataParallel(
+                copy.deepcopy(DDP_NET).cuda(),
```

Comment:
Shall we test parity for more complicated model such as torchvision.models.resnet50?

Reply:
I am afraid this will only work if torchvision is installed.

Reply:
yep, if you want, you can gate it with `HAS_TORCHVISION` flag.